chapters/resampling.qmd

```diff
@@ -263,6 +261,327 @@ mc_bias + mc_prec +
+
+For this simulation, the properties are about the same when MCCV and 10-fold CV are matched in terms of number of resamples and the amount allocated to the assessment set. 
+
+Our recommendation is to almost always use _V_ = 10. The bias and variance gains are both good with 10 folds. Reducing bias in 5-fold cross-validation is difficult but the 10-fold precision can be improved by increased replication. 
+
+## The Bootstrap {#sec-bootstrap}
+
+The bootstrap [@Efron1979boot;@davison1997bootstrap;@efron2003second;@EfronHastie2016a] is a resampling methodology originally created to compute the sampling distribution of statistics using minimal probabilistic assumptions^[This application will be discussed in section TODO.]. In this chapter, we'll define a bootstrap sample and show how it can be used to measure fit quality for predictive models. 
+
+For a training set with $n_{tr}$ data points, a bootstrap resample takes a random sample of the training set that is also size $n_{tr}$. It does this by sampling with replacement; when each of the $n_{tr}$ samples is drawn, it has no memory of the prior selections. This means that each row can be randomly selected again. For example, @fig-bootstrap-scheme shows another schematic for three bootstrap samples. In this figure, the thirteenth training set sample was selected to go into the first analysis set three times. For this reason, a bootstrap resample will contain multiple replicates of some training set points, while others will not be selected at all. The data that were never selected are used to create the assessment set.
+
+For the bootstrap, we'll refer to the mean estimate (line 12 in @alg-resampling) as the "ordinary" estimator (others will follow shortly). 
+
+```{r}
+#| label: fig-bootstrap-scheme
+#| echo: false
+#| out-width: '70%'
+#| fig-cap: "A schematic of three bootstraps resamples created from an initial pool of 30 data points."
+knitr::include_graphics("../premade/bootstraps.svg")
+```
+
+The probability that a data point will be picked is $1 / n_{tr}$ and, from this, the probability that a training set point is _not selected at all_ is 
+
+$$\prod_{i=1}^{n_{tr}} \left(1 - \frac{1}{n_{tr}}\right) = \left(1 - \frac{1}{n_{tr}}\right)^{n_{tr}}\approx e^{-1} = 0.368$$
+
+Since the bootstrap sample contains the _selected_ data, each training point is selected with probability $1 - e^{-1} \approx `r round((1 - exp(-1)), 3)`$. The implication is that, on average, the analysis set, contains about `r round((1 - exp(-1)) * 100, 1)`% unique training set points and the assessment set includes, on average, 36.8% of the training data. If we were to compare the amount of unique training set points held out by the bootstrap to V-fold cross-validation, the bootstrap is comparable to using $V = 3$. 
```

Comment:
```suggestion
Since the bootstrap sample contains the _selected_ data, each training point is selected with probability $1 - e^{-1} \approx `r round((1 - exp(-1)), 3)`$. The implication is that, on average, the analysis set, contains about `r round((1 - exp(-1)) * 100, 1)`% unique training set points and the assessment set includes, on average, 36.8% of the training data.  When comparing the number of unique training set points excluded by the bootstrap method to those excluded by V-fold cross-validation, the bootstrap method is roughly equivalent to using $V=3$ in cross-validation. 
```